core/modules/user/tests/user.test

```diff
@@ -2055,168 +1875,513 @@ class UserUserSearchTestCase extends BackdropWebTestCase {
+    $this->userLoadAndCheckRoleAssigned($account, $role_name, FALSE);
+  }
+
+  /**
+   * Check role on user object.
+   *
+   * @param object $account User.
+   * @param integer $role_name Role name.
+   * @param bool $is_assigned True if the role should present on the account.
+   */
+  private function userLoadAndCheckRoleAssigned($account, $role_name, $is_assigned = TRUE) {
+    $account = user_load($account->uid, TRUE);
+    if ($is_assigned) {
+      $this->assertTrue(in_array($role_name, $account->roles), 'The role is present in the user object.');
+    }
+    else {
+      $this->assertFalse(in_array($role_name, $account->roles), 'The role is not present in the user object.');
+    }
+  }
+}
+
+/**
+ * Tests user_validate_current_pass on a custom form.
+ */
+class UserValidateCurrentPassCustomForm extends BackdropWebTestCase {
+  /**
+   * User with permission to view content.
+   */
+  protected $accessUser;
+
+  /**
+   * User permission to administer users.
+   */
+  protected $adminUser;
+
+  function setUp() {
+    parent::setUp('user_form_test');
+    // Create two users
+    $this->accessUser = $this->backdropCreateUser(array('access content'));
+    $this->adminUser = $this->backdropCreateUser(array('administer users'));
+  }
+
+  /**
+   * Tests that user_validate_current_pass can be reused on a custom form.
+   */
+  function testUserValidateCurrentPassCustomForm() {
+    $this->backdropLogin($this->adminUser);
+
+    // Submit the custom form with the admin user using the access user's password.
+    $edit = array();
+    $edit['user_form_test_field'] = $this->accessUser->name;
+    $edit['current_pass'] = $this->accessUser->pass_raw;
+    $this->backdropPost('user_form_test_current_password/' . $this->accessUser->uid, $edit, t('Test'));
+    $this->assertText(t('The password has been validated and the form submitted successfully.'));
+  }
+}
+
+/**
+ * Test user entity callbacks.
+ */
+class UserEntityCallbacksTestCase extends BackdropWebTestCase {
+  function setUp() {
+    parent::setUp('user');
+
+    $this->account = $this->backdropCreateUser();
+    $this->anonymous = backdrop_anonymous_user();
+  }
+
+  /**
+   * Test label callback.
+   */
+  function testLabelCallback() {
+    $this->assertEqual(entity_label('user', $this->account), $this->account->name, t('The username should be used as label'));
+
+    // Setup a random anonymous name to be sure the name is used.
+    $name = $this->randomName();
+    config_set('system.core', 'anonymous', $name);
+    $this->assertEqual(entity_label('user', $this->anonymous), $name, t('The variable anonymous should be used for name of uid 0'));
+  }
+
+  /**
+   * Test URI callback.
+   */
+  function testUriCallback() {
+    $uri = entity_uri('user', $this->account);
+    $this->assertEqual('user/' . $this->account->uid, $uri['path'], t('Correct user URI.'));
+  }
+<<<<<<< HEAD
+}
+=======
```

Comment:
A couple of merge problems here are causing parse errors when running the tests.